src/main/php/CodacyPDepend/index.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace CodacyPDepend;
+
+require_once __DIR__ . '/../../../../vendor/autoload.php';
+require_once('CodacyConfiguration.php');
+require_once('CodacyReportGenerator.php');
+require_once('CodacyResult.php');
+
+use CodacyResult;
+use PDepend\Application;
+use Ds\Map;
+use LineComplexity;
+
+function genericMap($f, $list)
```

Comment:
I'm not happy on having to re-write std lib, have you had a look to something like:
https://www.php.net/manual/en/class.splobjectstorage.php ?

Reply:
I did this because `ASTArtifactList` is a custom iterable from pdepend.
I changed it to a simple foreach now.